mir_eval/display.py

```diff
@@ -0,0 +1,672 @@
+
+    Parameters
+    ----------
+    intervals_hier : list of np.ndarray
+        A list of segmentation intervals.  Each element should be
+        an n-by-2 array of segment intervals, in the format returned by
+        :func:`mir_eval.io.load_intervals` or
+        :func:`mir_eval.io.load_labeled_intervals`.
+        Segmentations should be ordered by increasing specificity.
+
+    labels_hier : list of list-like
+        A list of segmentation labels.  Each element should
+        be a list of labels for the corresponding element in
+        `intervals_hier`.
+
+    levels : list of string
+        Each element `levels[i]` is a label for the `i`th segmentation.
+        This is typically used to denote the levels in a segment hierarchy.
+
+    kwargs:
+        Additional keyword arguments to `labeled_intervals`.
+
+    Returns
+    -------
+    ax
+        A handle to the matplotlib axes
+    '''
+
+    # This will break if a segment label exists in multiple levels
+    if levels is None:
+        levels = list(range(len(intervals_hier)))
+
+    # Get the axis handle up front
+    kwargs.setdefault('ax', plt.gca())
+    ax = kwargs['ax']
+
+    # Count the pre-existing patches
+    n_patches = len(ax.patches)
+
+    for ints, labs, key in zip(intervals_hier[::-1],
+                               labels_hier[::-1],
+                               levels[::-1]):
+        labeled_intervals(ints, labs, label=key, **kwargs)
+
+    # Reverse the patch ordering for anything we've added.
+    # This way, intervals are listed in the legend from top to bottom
+    ax.patches[n_patches:] = ax.patches[n_patches:][::-1]
+    return kwargs['ax']
+
+
+def pitch(times, frequencies, midi=False, unvoiced=False, ax=None, **kwargs):
+    '''Visualize pitch contours
+
+    Parameters
+    ----------
+    times : np.ndarray, shape=(n,)
+        Sample times of frequencies
+
+    frequencies : np.ndarray, shape=(n,)
+        frequencies (in Hz) of the pitch contours.
+        Voicing is indicated by sign (positive for voiced,
+        non-positive for non-voiced).
+
+    midi : bool
+        If `True`, plot on a MIDI-numbered vertical axis.
+        Otherwise, plot on a linear frequency axis.
+
+    unvoiced : bool
+        If `True`, unvoiced pitch contours are plotted and indicated
+        by transparency.
+
+        Otherwise, unvoiced pitch contours are omitted from the display.
+
+    ax : matplotlib.pyplot.axes
+        An axis handle on which to draw the intervals.
+        If none is provided, a new set of axes is created.
+
+    kwargs :
+        Additional keyword arguments to `labeled_intervals`.
+
+    Returns
+    -------
+    ax
+        Handle to the plotting axes
+    '''
+
+    if ax is None:
+        ax = plt.gca()
+
+    # First, segment into contiguously voiced contours
+    frequencies, voicings = freq_to_voicing(np.asarray(frequencies,
+                                                       dtype=np.float))
```

Comment:
Yes, but this implementation isn't wrong, and it's a bit more forgiving anyway, so I see no reason to change it.